### ISSUE TITLE
QT Script (deprecated), but qtsetup asks for it

### ISSUE THREAD

**Issue by joeblew99** [therecipe/qt#351]
Just wanted to tidy up this...

Currently qtsetup looks for the qtscript folder in the QT installed folder, yet its is deprecated for QT 5.9. It maybe that the qtsetup and other bits need it still ?
QTscript is 810 MB on disk too ! GEEZ !

QT Maintenance tool GUI, shows it as deprecated..
![qtmaintenance](https://user-images.githubusercontent.com/11999859/26928211-8e747822-4c55-11e7-8be8-5c7806462be4.png)

````

x-MacBook-Pro:go apple$ qtsetup
INFO[0000] running: 'qtsetup prep'                      
INFO[0000] successfully created qtrcc symlink in your PATH (/usr/local/bin/qtrcc) 
INFO[0000] successfully created qtmoc symlink in your PATH (/usr/local/bin/qtmoc) 
INFO[0000] successfully created qtminimal symlink in your PATH (/usr/local/bin/qtminimal) 
INFO[0000] successfully created qtdeploy symlink in your PATH (/usr/local/bin/qtdeploy) 
INFO[0000] running: 'qtsetup check darwin' [docker=false] 
INFO[0000] GOOS:                        'darwin'        
INFO[0000] GOARCH:                      'amd64'         
INFO[0000] GOVERSION:                   'go1.8.3'       
INFO[0000] GOROOT:                   *  '/usr/local/opt/go/libexec' 
INFO[0000] GOPATH:                   *  '/Users/apple/workspace/go' 
INFO[0000] GOBIN:                       '/Users/apple/workspace/go/bin' 
INFO[0000] QT_HASH:                     'a5c03d787a4b94be671a6c9e4a80021ba6ec36bf' 
INFO[0000] QT_VERSION:               *  '5.9'           
INFO[0000] QT_VERSION_MAJOR:            '5.9'           
INFO[0000] QT_DIR:                   *  '/Users/apple/Qt' 
INFO[0000] QT_STUB:                     'false'         
INFO[0000] QT_DEBUG:                    'false'         
INFO[0000] QT_QMAKE_DIR:                ''              
INFO[0000] QT_WEBKIT:                   'false'         
INFO[0000] QT_HOMEBREW:                 'false'         
INFO[0000] XCODE_DIR:                   '/Applications/Xcode.app' 
INFO[0000] running: 'qtsetup generate darwin' [docker=false] 
ERRO[0004] failed to load /Users/apple/Qt/Docs/Qt-5.9/qtscript/qtscript.index  error="open /Users/apple/Qt/Docs/Qt-5.9/qtscript/qtscript.index: no such file or directory"
WARN[0004] parser.LoadModule                             error=EOF module=Script
ERRO[0008] failed to load /Users/apple/Qt/Docs/Qt-5.9/qtscripttools/qtscripttools.index  error="open /Users/apple/Qt/Docs/Qt-5.9/qtscripttools/qtscripttools.index: no such file or directory"
WARN[0008] parser.LoadModule                             error=EOF module=ScriptTools
INFO[0010] generating full qt/core                      
INFO[0021] generating full qt/androidextras             
INFO[0021] generating full qt/gui                       
INFO[0034] generating full qt/network                   
INFO[0040] generating full qt/xml                       
INFO[0043] generating full qt/dbus                      
INFO[0046] generating full qt/nfc                       
INFO[0049] generating full qt/script                    
ERRO[0051] failed to run command                         cmd="/Users/apple/Qt/5.9/clang_64/bin/qmake /Users/apple/workspace/go/src/github.com/therecipe/qt/script.pro -spec macx-clang CONFIG+=x86_64 CONFIG+=release" env= error="exit status 3" func=RunCmdOptional name="run qmake for darwin on darwin"
Info: creating stash file /Users/apple/workspace/go/src/github.com/therecipe/qt/script/.qmake.stash
Project ERROR: Unknown module(s) in QT: script

ERRO[0051] failed to load /Users/apple/workspace/go/src/github.com/therecipe/qt/script/Makefile  error="open /Users/apple/workspace/go/src/github.com/therecipe/qt/script/Makefile: no such file or directory"
INFO[0051] generating full qt/sensors                   
INFO[0056] generating full qt/positioning               
INFO[0059] generating full qt/widgets                   
INFO[0087] generating full qt/sql                       
INFO[0091] generating full qt/macextras                 
INFO[0094] generating full qt/qml                       
INFO[0098] generating full qt/websockets                
INFO[0101] generating full qt/xmlpatterns               
INFO[0104] generating full qt/bluetooth                 
INFO[0108] generating full qt/webchannel                
INFO[0110] generating full qt/svg                       
INFO[0114] generating full qt/multimedia                
^C


````

**Comment by joeblew99** [comment]
So i installed qtscript with the QT maintenance GUI, and all good.
But now a weird issue with QT 5.9 maybe:

``` qtsetup ```  runs clean. It compiled and opened the test apps.

But then i tried to build the qml/drawer_x example.
```
x-MacBook-Pro:drawer_nav_x apple$ pwd
/Users/apple/workspace/go/src/github.com/therecipe/qt/internal/examples/qml/drawer_nav_x
x-MacBook-Pro:drawer_nav_x apple$ qtdeploy
````

Fails. Mhhh.

so i tried ``` qtsetup test ```, and it now fails.

Is QT 5.9 NOT supported yet maybe ?

**Comment by therecipe** [comment]
Hey

Yeah, you can omit the Script module if you don't need it.
The setup and qtdeploy should keep working normally, but there will be warnings that the script module is not found. I was planning to remove the use of script module once they stop distributing it with their installers, but it's now deprecated since 5.4 or so and still there :D

Yeah, 5.9 support is still a bit buggy even through the desktop targets should work already.
I already made much of the changes necessary, but then figured that they also changed the form of their pre-built packages and thereby broke the CI and the docker images.
So I it will take a while until 5.9 is fully supported, but I will cherry pick most of the other fixes tomorrow and push them up.

**Comment by joeblew99** [comment]
Ok thanks and makes sense.

5.9 is a long term release too I noticed so hopefully by sticking to that for a while it will lower your flux and stabilise the golang code. So can work on other compelling aspects we have discussed.

I have an enterprise evaluation licence and will test signing of android and iOS apps.
Is there anything else you want me to test just tell.
I will use drawer-x as the test case for now.

Eager to get some 2d and 3d graphics demos into testing.
I have one of each written in CPp / qt, and porting to golang is coming along so we can put them into the demo area if your up for it ?
